### DISCUSSION TITLE
Integration with git

### DISCUSSION THREAD

**Discussion by RandomGamingDev** [processing/p5.js-web-editor#2671] [General]
### Increasing Access

Integration to git would allow for things like easier backing up, and sharing of projects in a way that allows them to be easily modified by everyone. (This is basically https://github.com/processing/p5.js-web-editor/issues/753, but with the added features of being able to write to git repositories.)
Currently, p5.js makes it hard to collaborate with others since projects are in a singular account and can't be shared between multiple, and there's also no version control system, blame system, or other system for easy restoration and exploration when dealing with projects that could a complex branch commit system instead of the current p5.js system, which also means that adding this would fix a large portion of p5.js's major problems.
This could be done using something like [Isomorphic Git](https://isomorphic-git.org/)

### Feature request details

Add the ability to use git's features via some sort of UI, or CLI made specifically for git.
This could be done using something like [Isomorphic Git](https://isomorphic-git.org/)

**Comment by jht9629-nyu** [comment]
This would be a great feature.
Currently I've use a collection of nodejs scripts to batch download my work from p5js editor and backup in a git repo.
https://github.com/jht9629-nyu/my-p5js-repo-2023
I work mostly offline in visual studio code and then copy the scripts manually back into p5js editor to share with my students.
This feature would speed up my workflow and provide a gently migration out of and back into p5js editor

**Comment by jht9629-nyu** [comment]
I've update my scripts to download and manage your p5 projects (sketch.js + media):
- https://github.com/molab-itp/p5mirror

this repo replaces repo _my-p5js-repo-2023_ mentioned earlier:

The repo p5mirror is in need of more first-user friendly documentation. 
I hope experienced coders will be able to follow the current documentation.

If you are comfortable with the command line, try the usage documentation and let me know how it goes
- https://github.com/molab-itp/p5mirror?tab=readme-ov-file#usage

**Any feedback and/or collaboration is appreciated**

**Reply by jht9629-nyu** [reply]
Update to 
- https://github.com/molab-itp/p5mirror

**Comment by p5moLab** [comment]
CRUD support added, with some limitations
to allow you to create-read-write-delete your editor.p5js sketches
https://github.com/molab-itp/p5mirrorLib?tab=readme-ov-file#crud
